package/scie-pants.toml

```
@@ -218,6 +218,8 @@ args = [
     "{scie.env.PANTS_DEBUG}",
     "--debugpy-requirement",
     "{scie.env.PANTS_DEBUGPY_VERSION}",
+    "--pants-pex-release-url-base",
+    "{scie.env.SCIE_PANTS_PANTS_PEX_RELEASE_URL_BASE}",
```

Does it have to be this value, or could we collapse the doubled `PANTS` in there.. ?

```suggestion
    "{scie.env.SCIE_PANTS_PEX_RELEASE_URL_BASE}",
```

---

I think we get to decide. I see this as an env varname equivalent of something like `scie_pants::pants_pex::release_url_base`

---

Yea, I see the pattern it comes from. I'm just asking if we couldn't fold the double pants, much like you'd fold tripple consonants into just two when you write together two words. Seeing the namespaces are kind of loosely related to the name and not enforced.. 👀  I'm in favour of brevity/readability here..

---

Oh the other ambuguity is `scie-pants` does also download Pex from GitHub: https://github.com/pantsbuild/scie-pants/blob/891972ba3cb75f2bb7fe383927f3f25822e4c826/package/pbt.toml#L24

:grimacing: So the ambiguity would be is if this was "scie-pants PEX release url base" or "scie pants-PEX release url base"

---

(Not saying I love the double-Pants)

---

We could also try out other names. My brain isn't coming up with much...

---

Oh, right. Yea I think it may be a mistake to include PEX in the var name here (took me until now to realize exactly what it all meant.. granted my head is extra slow today but still).

Equally, scie isn't really interesting either. Why not attempt to get at what it is for only: `PANTS_BOOTSTRAP_BASE_URL` or some such..?

---

then comes the considerations of the namespaces that squats on (the BOOTSTRAP, in this case)

---

I'm worried about conflict with PANTS_ env vars from options

---

maybe `PANTS_INSTALL_BASE_URL` ?

---

> I'm worried about conflict with PANTS_ env vars from options

yep, that was my thought two seconds after posting too... but if we can get out of that?

---

`DOWNLOAD_PANTS_BASE_URL` 🤪

---

Oooh a bikeshed, let me join in!

I like the `PANTS_BOOTSTRAP_...` idea from @kaos : for better or for worse, scie-pants already uses that namespace, so seems sensible to continue using that (i.e. conflicts with `PANTS_...` env vars already already a thing).

These env vars already exist:

- `PANTS_BOOTSTRAP_GITHUB_API_BEARER_TOKEN`
- `PANTS_BOOTSTRAP_TOOLS`
- `PANTS_BOOTSTRAP_VERSION`

(There's one reference to `PANTS_BOOTSTRAP_URLS` too, but I don't know what reads it)

So, `PANTS_BOOTSTRAP_BASE_URL` or some variation sounds good to me!

---

Ah, actually: `PANTS_BOOTSTRAP_URLS` is referenced/described in https://github.com/pantsbuild/scie-pants?tab=readme-ov-file#features

So, maybe something could be riffed on there.

(Also, those docs should be updated in this PR, as appropriate.)

---

Ohh ok we're getting somewhere I think, that's really good info @huon

PANTS_BOOTSTRAP_PANTS_PEX_BASE_URL?

---

> So, maybe something could be riffed on there.

Hmm yeah let me try that :+1:

---

DONE! :rocket: